python/src/xspress_detector/control/fp_xspress_adapter.py

```diff
@@ -177,6 +180,20 @@ def setup_rank(self):
                 logging.debug(OdinDataAdapter.ERROR_FAILED_TO_SEND)
                 logging.error("Error: %s", err)
 
+    def configure_dataset_offset(self, offset):
+        for client in self._clients:
+            try:
+                parameters = {
+                    "xspress": {
+                        "offset": int(offset),
+                    },
+                }
+                logging.warning("Sending: {}".format(parameters))
+                client.send_configuration(parameters)
+            except Exception as err:
+                logging.debug(OdinDataAdapter.ERROR_FAILED_TO_SEND)
+                logging.error("Error: %s", err)
+
```

Comment:
Is this needed? It doesn't have any special logic like `configure_dataset_offset`, so can't it just be handled by the default case [here](https://github.com/DiamondLightSource/xspress-detector/pull/1/files#diff-9a5f4d7b893fdaed32ac10f3a5b6c09e988f8c2593f44ea44ad22d1d2f86234fR156)?

Reply:
Hi Gary, do you mean if the whole function is needed or just the exception bit?

Reply:
I mean the whole function. Why does it need to intercept `offset/offset_adjustment` and send it to `xspress/offset` instead? Is this so that it can re-use the same PV for something different?

Reply:
Yes, basically I'm using the same PV that's used by the Offset plugin and inside the Xspress FrameProcessorPlugin code I [use it as a variable](https://github.com/DiamondLightSource/xspress-detector/blob/pause-dev/cpp/data/frameProcessor/src/XspressProcessPlugin.cpp#L229) that's responsible for [offsetting the meta data](https://github.com/DiamondLightSource/xspress-detector/blob/pause-dev/cpp/data/frameProcessor/src/XspressProcessPlugin.cpp#L518) sent to the Meta Writer.
I thought this would be a nicer way of doing it instead of creating a completely new PV just for that purpose.

Reply:
OK, I think that is fine. Shouldn't it also send the config to the original parameter too, or is it not supposed to be the same? That is a bit confusing if not.

Reply:
Yeah, I think it's a bit confusing because even though we're using the configurations that are meant for a dataset (in this case the data dataset) that original dataset is not used due to the way Xspress is writting it's data (using the raw datasets for each channel), so in the end I'm just intercepting the values but passing it to the individual FPs to apply it in their datasets that contain the real data, but it doesn't make any difference if the original parameter receives the data or not.

Reply:
OK. I think this is fine for now. FastCS will allow us to do this in a much more sane way and then some of this complication can be removed.